api/queries/user/githubProfile.js

```diff
@@ -1,10 +1,50 @@
 // @flow
+require('now-env');
 import type { DBUser } from 'shared/types';
+import cache from 'shared/cache/redis';
+import fetch from 'isomorphic-fetch';
 
-export default ({ githubProviderId, githubUsername }: DBUser) => {
-  if (!githubProviderId || !githubUsername) return null;
-  return {
-    id: githubProviderId,
-    username: githubUsername,
-  };
+export default async ({ githubProviderId, id }: DBUser) => {
+  if (!githubProviderId) return null;
+
+  const cachedGithubProfile = await cache.get(`githubProfile:${id}`);
+
+  if (cachedGithubProfile) {
+    const parsed = JSON.parse(cachedGithubProfile);
```

Comment:
nit: `try/catch` this JSON.parse and, if it errors, go fetch it again to make sure bad data in the cache doesn't hurt users.

Probably something like:

```JS
const cachedGitHubProfile = ...

if (cachedGitHubProfile) {
  try {
    const parsed = JSON.parsed(cachedGitHubProfile);
    return { ... };
  // Just ignore errors, causing the `return` not to run and skipping below the if block
  } catch(err) {}
}

// ...other code here
const apiUrl = ...;
```